### DISCUSSION TITLE
Is Savedata information in tasmota docs ok?

### DISCUSSION THREAD

**Discussion by ortegafernando** [arendst/Tasmota#14880] [Q&A] [answered]
Hi, I have read a lot of issues with Savedata ON and OFF

But tasmota docs:
```
SaveData
0 = save parameter changes only manually, e.g. with Restart 1
1 = save parameter changes every second (default)
2..3600 = save parameter changes every x second
```
What is really this? I have read that you can put SaveData OFF not to write flash memory everytime, but tasmota docs dont seem to explain this well.

Thanks.

**Comment by ascillato2** [comment, marked as answer]
`savedata 1` or `savedata ON`: Tasmota will check for settings changed every second. if there is a change (like a relay turning ON) it will be saved after 1 second of the event.

`savedata 100`: Tasmota will check for settings changed every 100 seconds. if there is a change (like a relay turning ON) it will be saved between 100 second of the event.

`savedata 0` or `savedata OFF`: Tasmota will not check for any change in settings, so the new settings will be saved ONLY if you manually restart the device, for example by command `restart 1` or by **_restart_** in the WebUI. If you disconnect the power of the device without restarting it first, your new settings won't be saved in flash.

**Reply by sfromis** [reply]
For many settings, OFF is a synonym for 0 and ON is a synonym for 1. This means that `SaveData OFF` is the same as `SaveData 0`. Note that when it comes to what is saved, in some cases `SetOption0` may be more practical than `SaveData OFF`.

https://tasmota.github.io/docs/Commands/#how-to-use-commands explains:
> Instead of 0 you can use off or false and instead of 1 you can use on or true.

**Reply by ortegafernando** [reply]
> `savedata 1` or `savedata ON`: Tasmota will check for settings changed every second. if there is a change (like a relay turning ON) it will be saved after 1 second of the event.
> 
> `savedata 100`: Tasmota will check for settings changed every 100 seconds. if there is a change (like a relay turning ON) it will be saved between 100 second of the event.
> 
> `savedata 0` or `savedata OFF`: Tasmota will not check for any change in settings, so the new settings will be saved ONLY if you manually restart the device, for example by command `restart 1` or by **_restart_** in the WebUI. If you disconnect the power of the device without restarting it first, your new settings won't be saved in flash.

for example savedata 100 or savedata 1000 takes sense when we use setoption0 0, don't it?

**Reply by ascillato** [reply]
> for example savedata 100 or savedata 1000 takes sense when we use setoption0 0, don't it?

mmm, not exactly.

Setoption0 is only for the status of relays. The rest of parameters (dimming level, color, etc) are saved depending on the time set in SAVEDATA.

**Reply by ortegafernando** [reply]
Great answers, thanks everybody.

**Reply by ortegafernando** [reply]
> `savedata 1` or `savedata ON`: Tasmota will check for settings changed every second. if there is a change (like a relay turning ON) it will be saved after 1 second of the event.

Just to ensure that, savedata "applies" to all parameters but relays' ones.
Relays's parameters are controlled by SetOption0.

Is it ok?

**Reply by ascillato** [reply]
SAVEDATA is the time when ALL parameters are going to be saved.

SETOPTION0 is just if the relay status are going to be saved or not. If they are saved, they will be saved at SAVEDATA time. If SETOPTION0 is 0, no matter what parameter SAVEDATA has, relays WILL NOT be saved.

**Reply by sfromis** [reply]
`SaveData` applies to full setup including Power states. However, the latter can be excluded using `SetOption0`. While this sounds like what you write, the difference is that without `SaveData` being allowed, power states are not saved, irrespective of `SetOption0`.

**Reply by quantenschaum** [reply]
What exactly does `savedata` apply to, anything that is stored in the flash? It affects storing of relay state and dimmer state, does it also apply to counter values and any other config parameters like `deepsleeptime` or power calibration?

**Reply by sfromis** [reply]
It applies to everything persistent, including all that you mention, unless specifically disabled. Saving of settings data is however not triggered every time counters or energy monitoring data is updated, but only if saving for other reasons, including automatic save at midnight local time.

Note that a few features do have settings not persistent, needing to happen after each boot, but this is the exception.
